src/manifolds/SymmetricPositiveDefiniteFixedDeterminant.jl

```diff
@@ -0,0 +1,152 @@
+@doc raw"""
+    SymmetricPositiveDefiniteFixedDeterminant{N,D} <: AbstractDecoratorManifold{ℝ}
+
+The manifold of symmetric positive definite matrices of fixed determinant ``d > 0``, i.e.
+
+````math
+\mathcal P_d(n) =
+\bigl\{
+p ∈ ℝ^{n × n} \ \big|\ a^\mathrm{T}pa > 0 \text{ for all } a ∈ ℝ^{n}\backslash\{0\}
+  \text{ and } \det(p) = d
+\bigr\}.
+````
+This manifold is modelled as a submanifold of [`SymmetricPositiveDefinite`](@ref)`(n)`.
+
+The tangent space is modelled the same as for [`SymmetricPositiveDefinite`](@ref)`(n)`
+and consists of all symmetric matrices with zero trace
+```math
+    T_p\mathcal P_d(n) =
+    \bigl\{
+        X \in \mathbb R^{n×n} \big|\ X=X^\mathrm{T} \text{ and } \operatorname{tr}(p) = 0
+    \bigr\},
+```
+since for a constant determinant we require that `0 = D\det(p)[Z] = \det(p)\operatorname{tr}(p^{-1}Z)` for all tangent vectprs ``Z``.
+Additionally we store the tangent vectors as `X=p^{-1}Z`, i.e. symmetric matrices.
+
+# Constructor
+
+    SymmetricPositiveDefiniteFixedDeterminant(n::Int, d::Real=1.0)
+
+generates the manifold $\mathcal P_d(n) \subset \mathcal P(n)$ of determinant ``d``,
+which defaults to 1.
+"""
+struct SymmetricPositiveDefiniteFixedDeterminant{N,TD<:Real} <: AbstractDecoratorManifold{ℝ}
+    d::TD
+end
+
+function SymmetricPositiveDefiniteFixedDeterminant(n::Int, d::F=1.0) where {F<:Real}
+    @assert d > 0 "The determinant has to be positive but was provided as $d."
+    return SymmetricPositiveDefiniteFixedDeterminant{n,F}(d)
+end
+
+function active_traits(f, ::SymmetricPositiveDefiniteFixedDeterminant, args...)
+    return merge_traits(IsEmbeddedSubmanifold())
+end
+
+@doc raw"""
+    check_point(M::SymmetricPositiveDefiniteFixedDeterminant{n}, p; kwargs...)
+
+Check whether `p` is a valid manifold point on the [`SymmetricPositiveDefiniteFixedDeterminant`](@ref)`(n,d)` `M`, i.e.
+whether `p` is a [`SymmetricPositiveDefinite`](@ref) matrix (checked by traits) ofsize `(n,n)`
+
+with determinant ``\det(p) = d``.
+
+The tolerance for the determinant of `p` can be set using `kwargs...`.
+"""
+function check_point(
+    M::SymmetricPositiveDefiniteFixedDeterminant{n},
+    p;
+    kwargs...,
+) where {n}
+    if det(p) != M.d
```

Comment:
```suggestion
    if det(p) ≉ M.d
```
I think approximate equality should be sufficient.